clicommand/pipeline_upload.go

```diff
@@ -398,3 +462,22 @@ func searchForSecrets(l logger.Logger, cfg *PipelineUploadConfig, environ map[st
 		}
 	}
 }
+
+func validateJWK(key jwk.Key, alg jwa.SignatureAlgorithm) error {
+	validKeyTypes := []jwa.KeyType{jwa.RSA, jwa.EC, jwa.OctetSeq}
+	if !slices.Contains(validKeyTypes, key.KeyType()) {
+		return fmt.Errorf("unsupported key type %s. Key type must be one of %v", key.KeyType(), validKeyTypes)
+	}
+
+	validAlgsForType := map[jwa.KeyType][]jwa.SignatureAlgorithm{
```

Comment:
Add a note in a comment that this list deliberately excludes PKCS 1.5 (because it is discouraged, etc)

And probably also worth a note that we're waiting on the library to support Ed25519 (or whatever the JOSE folks are calling it)

Reply:
so it turns out that i had intended (at one point?) to not include RSA-PKCSv1.5 in this list, but it's totally in there at the moment - i'll clean it out. see my comment above about Ed25519 - it's very doable!